astropy/coordinates/tests/test_representation.py

```diff
@@ -631,6 +702,63 @@ def test_initialize_with_nan(self):
         assert_array_equal(np.isnan(psr.theta), [True, False])
         assert_array_equal(np.isnan(psr.r), [False, True])
 
+    def test_transform(self):
+
+        s1 = PhysicsSphericalRepresentation(
+            phi=[1, 2] * u.deg, theta=[3, 4] * u.deg, r=[5, 6] * u.kpc)
+
+        matrix = rotation_matrix(-10, "z", u.deg)
+
+        s2 = s1.transform(matrix)
+
+        assert_allclose_quantity(s2.phi, s1.phi + 10 * u.deg)
+        assert_allclose_quantity(s2.theta, s1.theta)
+        assert_allclose_quantity(s2.r, s1.r)
+
+        # now with a non rotation matrix
+        matrix = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+        s3 = s1.transform(matrix)
+        expected = (s1.to_cartesian().transform(matrix)
+                      .represent_as(PhysicsSphericalRepresentation))
+
+        assert_allclose_quantity(s3.phi, expected.phi)
+        assert_allclose_quantity(s3.theta, expected.theta)
+        assert_allclose_quantity(s3.r, expected.r)
+
+    def test_transform_with_NaN(self):
+        # all over again, but with a NaN in the distance
+
+        s1 = PhysicsSphericalRepresentation(
+            phi=[1, 2] * u.deg, theta=[3, 4] * u.deg, r=[5, np.nan] * u.kpc)
+
+        matrix = rotation_matrix(-10, "z", u.deg)
+
+        s2 = s1.transform(matrix)
+
+        assert_allclose_quantity(s2.phi, s1.phi + 10 * u.deg)
+        assert_allclose_quantity(s2.theta, s1.theta)
+        assert_allclose_quantity(s2.r, s1.r)
+
+        # now with a non rotation matrix
+        matrix = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+
+        s3 = s1.transform(matrix)
+        # s3 should not propagate Nan.
+        assert tuple(np.isnan(s3.phi.deg)) == (False, False)
```

Comment:
As in the code on line 702, I think `assert_array_equal` makes a bit more sense here - what is the `tuple` doing??

Reply:
I can change that. The tuple was a quick and dirty element-wise comparison.